src/sentry/integrations/services/integration/impl.py

```diff
@@ -354,6 +355,57 @@ def update_organization_integration(
         )
         return ois[0] if len(ois) > 0 else None
 
+    def start_grace_period_for_provider(
+        self,
+        *,
+        organization_id: int,
+        provider: str,
+        grace_period_end: datetime,
+        status: int = ObjectStatus.ACTIVE,
+        skip_oldest: bool = True,
+    ) -> list[RpcOrganizationIntegration]:
+        current_org_ois = OrganizationIntegration.objects.filter(
+            organization_id=organization_id,
+            integration__provider=provider,
+            status=status,
+        )
+        ois_to_update = list(current_org_ois.values_list("id", flat=True))
+
+        if skip_oldest:
+            # Get all associated OrganizationIntegrations for the Integrations used by this org
+            all_ois = (
+                OrganizationIntegration.objects.filter(
+                    integration__in=current_org_ois.values_list("integration_id", flat=True),
+                    status=status,
+                    integration__provider=provider,
+                )
+                .order_by("date_added")
```

Comment:
I think you can do `order_by("date_added", "date_updated")` so when you append them to the list they're already sorted the way you want (no need for additional sort on line 393). But maybe you can just order by `id` since it's always in increasing order?